### ISSUE TITLE
KeyError when using root-level `Fn::Transform` is Resources block

### ISSUE THREAD

**Issue by jmalins** [aws/aws-sam-cli#2206]
### Description ###

https://github.com/aws/aws-sam-cli/blob/0b22b1725a540a34b5d981d062edd1b138e245a1/samcli/lib/warnings/sam_cli_warning.py#L75

This code fails with a `KeyError` if a transform block is used to import CFN resources from an external file, as in the following example:
```yaml
Resources:
  Fn::Transform:
    Name: AWS::Include
    Parameters:
      Location: ./deploy/queries.yaml
```
The Fn::Transform tag does not have a `Type` key.

### Observed Behavior ###

I get the following error when packaging a template structured with the transform above:
```
  ...
  File "/usr/local/Cellar/aws-sam-cli/1.2.0/libexec/lib/python3.7/site-packages/samcli/lib/warnings/sam_cli_warning.py", line 53, in check_template_for_warning
    should_warn, warning_message = warning.check(template_dict)
  File "/usr/local/Cellar/aws-sam-cli/1.2.0/libexec/lib/python3.7/site-packages/samcli/lib/warnings/sam_cli_warning.py", line 74, in check
    for (_, resource) in template_dict.get("Resources", {}).items()
  File "/usr/local/Cellar/aws-sam-cli/1.2.0/libexec/lib/python3.7/site-packages/samcli/lib/warnings/sam_cli_warning.py", line 75, in <listcomp>
    if resource["Type"] == "AWS::Serverless::Function"
KeyError: 'Type'
```

### Expected Behavior ###

At worst, tags without a `Type` key in the resources block should be ignored. The more correct behavior is probably to apply this check after resolving transforms.

**Comment by jmalins** [comment]
For anyone receiving this error, I found a temporary workaround.

It appears CFN ignores additional properties added to the `Fn:Transform` tag. One can therefore do this:
```yaml
Resources:
  Fn::Transform:
    Name: AWS::Include
    Parameters:
      Location: ./deploy/queries.yaml
    Type: ignore  # extra property workaround sam-cli bug
```

**Comment by moelasmar** [comment]
This issue is fixed in Release V1.3.0
I will close this issue. You can reopen it if you still have any issue.
